app_dart/lib/src/service/scheduler.dart

```diff
@@ -885,12 +787,10 @@ $stackTrace
       sha: headSha,
     );
 
-    late CiYamlSet ciYaml;
-    if (commit.branch == Config.defaultBranch(commit.slug)) {
-      ciYaml = await getCiYaml(commit, validate: true);
-    } else {
-      ciYaml = await getCiYaml(commit);
-    }
+    final ciYaml = await _ciYamlFetcher.getCiYamlByDatastoreCommit(
+      commit,
+      validate: commit.branch == Config.defaultBranch(commit.slug),
+    );
```

Comment:
nit: Are these the only places we call validate: true? Maybe we just move that logic into the getCiYamlByDatastoreCommit()?


I think I see one place where we don't - but that feels like the exception.
```dart
// process checkrun
final ciYaml = await _ciYamlFetcher.getCiYamlByDatastoreCommit(
commit,
);
```

Reply:
Will revisit in a follow-up PR.